src/components/__tests__/change-view.test.jsx

```diff
@@ -0,0 +1,356 @@
+/* eslint-env jest */
+
+import React from 'react';
+import {shallow} from 'enzyme';
+import ChangeView, {defaultDiffType} from '../change-view/change-view';
+import layeredStorage from '../../scripts/layered-storage';
+import simplePage from '../../__mocks__/simple-page.json';
+import WebMonitoringDb from '../../services/web-monitoring-db';
+import {diffTypesFor} from '../../constants/diff-types';
+
+jest.mock('../../scripts/layered-storage');
+
+const mockChange = {
+  uuid: 'adc70e01-0723-4a28-a8e4-ca4b551ed2ae..0772dc83-7966-4881-8901-eceb051b9536',
+  uuid_from: 'adc70e01-0723-4a28-a8e4-ca4b551ed2ae',
+  uuid_to: '0772dc83-7966-4881-8901-eceb051b9536',
+  priority: null,
+  current_annotation: {},
+  created_at: null,
+  updated_at: null,
+  significance: null
+};
+
+// Note for testing relevant diffTypes:
+// the defaultDiffType (SIDE_BY_SIDE_RENDERED) is only relevant for the "text/html" media type
+// the relevant types for "text/*" are a subset of the relevant types for "text/html"
+// the relevant types for 'text/html' and "*/*" are mutually exclusive
+
+const diffTypeStorage = 'edgi.wm.ui.diff_type';
+
+describe('change-view', () => {
+  const mockApi = Object.assign(
+    Object.create(WebMonitoringDb.prototype),
+    {getChange: () => Promise.resolve(mockChange)}
+  );
+
+  describe('initial diffType', () => {
+    describe('when a diffType has been stored in layeredStorage', () => {
+      describe('when that diffType is relevant to the pages being compared', () => {
+        it('sets state.diffType to the stored value', () => {
+          const storedDiffType = 'CHANGES_ONLY_TEXT';
+          layeredStorage.getItem.mockReturnValue(storedDiffType);
```

Comment:
I think I missed this before, but I can imagine it might cause issues in the future. Since the collapsed view setting is now using layered storage, it’s also getting this value as its setting. This mock is way too broad.

I think we’re probably better off to either add a custom mock for LayeredStorage in the `__mocks__` directory or write one in this file where we call `jest.mock('../../scripts/layered-storage')` up on line 11. It could be as simple as:

```js
jest.mock('../../scripts/layered-storage', () => ({
  __esModule: true,
  default: {
    getItem (key) { return this._data[key]; },
    setItem (key, value) { return this._data[key] = value; },
    removeItem (key) { return delete this._data[key]; },
    clear () { this._data = {}; },
    _data: {}
  }
}));
```

And then have a teardown step for these tests that resets it:

```js
afterEach(() => {
  layeredStorage.clear();
});
```

And you’d use it at this line by doing:

```js
layeredStorage.setItem(diffTypeStorage, storedDiffType);
```